### PR TITLE
adding example program for interconnect-type resource

### PR DESCRIPTION
**Pull request by Sowmyahsg**
Description
Support for Interconnect Types for API 500 , 600 and 800

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/interconnect_type.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	ovc := ClientOV.NewOVClient(
+		os.Getenv("ONEVIEW_OV_USER"),
+		os.Getenv("ONEVIEW_OV_PASSWORD"),
+		os.Getenv("ONEVIEW_OV_DOMAIN"),
+		os.Getenv("ONEVIEW_OV_ENDPOINT"),
+		false,
+		800,
+		"*")
+
+	ovVer, _ := ovc.GetAPIVersion()
```
Comment (soodpr) [inline examples/interconnect_type.go]: no need to call getAPIversion here.

File: examples/interconnect_type.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	interconnect, err := ovc.GetInterconnectTypeByName(existing_interconnect)
+	if err != nil {
+		fmt.Println(err)
+	} else {
+		fmt.Println(interconnect)
+	}
+
+	interconnect_type_uri := utils.NewNstring("rest/interconnect-types/ed8fbf3b-1962-4602-8eeb-6ef2f303a67f")
+	fmt.Println("#................... Interconnect Type by Uri ....................#")
+	int_uri, err := ovc.GetInterconnectTypeByUri(interconnect_type_uri)
```
Comment (soodpr) [inline examples/interconnect_type.go]: variable name should be interconnect_type

File: examples/interconnect_type.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	if err != nil {
+		panic(err)
+	} else {
+		fmt.Println(int_uri)
+	}
+
+	sort := "name:desc"
+	interconnect_type_list, err := ovc.GetInterconnectTypes("", sort)
```
Comment (soodpr) [inline examples/interconnect_type.go]: i think you should add other parameters like start, count

File: examples/interconnect_type.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	if err != nil {
+		panic(err)
+	} else {
+		fmt.Println("# ................... Interconnect Type List .................#")
+		for i := 0; i < len(interconnect_type_list.Members); i++ {
+			fmt.Println(interconnect_type_list.Members[i].Name)
+		}
+	}
+}
```
Comment (jyothis-gireesan) [inline examples/interconnect_type.go]: Same as Priyankas

File: examples/interconnect_type.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+		fmt.Println(err)
+	} else {
+		fmt.Println(interconnect)
+	}
+
+	interconnect_type := utils.NewNstring("rest/interconnect-types/ed8fbf3b-1962-4602-8eeb-6ef2f303a67f")
+	fmt.Println("#................... Interconnect Type by Uri ....................#")
+	int_uri, err := ovc.GetInterconnectTypeByUri(interconnect_type)
+	if err != nil {
+		panic(err)
```
Comment (AnushaY1916) [inline examples/interconnect_type.go]: Panic is not required, you can just print the error and proceed further

File: examples/interconnect_type.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	} else {
+		fmt.Println(int_uri)
+	}
+
+	sort := "name:desc"
+	interconnect_type_list, err := ovc.GetInterconnectTypes("", "", "", sort)
+	if err != nil {
+		panic(err)
```
Comment (AnushaY1916) [inline examples/interconnect_type.go]: Same comment as above

**Pull request merged**